keyboards/handwired/ferris/matrix.c

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2013 Oleg Kostyuk <cub.uanic@gmail.com>
+          2020 Pierre Chevalier <pierrechevalier83@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*
+ * This code was heavily inspired by the ergodox_ez keymap, and modernized
+ * to take advantage of the quantum.h microcontroller agnostics gpio control
+ * abstractions and use the macros defined in config.h for the wiring as opposed
+ * to repeating that information all over the place.
+ */
+
+#include QMK_KEYBOARD_H
```

Comment:
I think the MCP init stuff in ferris.h should go in this file, so it's all in one place.

(Sorry, just taking the opportunity to figure out how MCP custom matrix should be done for future reference & refactoring)

Reply:
Don't apologize! I've been really appreciating this review, with the code getting simpler and simpler as we go. Totally up for laying good foundations for split designs with io expanders as I dig these. There is still some bitflip logic on the mcp side, btw as it was convenient with all the pins being sequentially index with  but in the longer term, I could totally see it going away in favour of logic similar to the mcp side. Once that happens, I could see most of the logic being generic and being able to move to split_common with only defines to assign the pins on each side needed for any new board. Small steps, though :smile: